### DISCUSSION TITLE
Intersection management service using CAM - Velocity & Acceleration

### DISCUSSION THREAD

**Discussion by BouchraBenkoussas** [riebl/artery#188] [Q&A]
Hello @riebl,

I would like to create a service to manage the turning manoeuvre at intersection. l'm trying to estimate the collision time using the CAM and I need to calculate the velocity and acceleration component. I found that I can get just the overall value of each of them. Would it be possible to get the two components (longitudinal and lateral) of each of them?

Thank you in advance and kind regards,
Bouchra.

**Comment by riebl** [comment]
As far as I know, SUMO does not report lateral acceleration. You can retrieve a vehicle's lateral speed, however, via `TraCIAPI::VehicleScope::getLateralSpeed`. Artery's `VehicleController` grants you access to the full TraCI API. You may manually derive the acceleration by observing the speed changes over time.

**Reply by BouchraBenkoussas** [reply]
Hi @riebl,

I tried to retrieve the lateral speed via `TraCIAPI::VehicleScope::getLateralSpeed` with this line of code `std::cout<<"getLateralSpeed: "<<api.vehicle().getLateralSpeed(mVehicleController->getVehicleId());` but the function returns a 0.
Meanwhile, via the getspeed I was able to get the right overall speed which corresponds to the right vehicle id 
`std::cout<<"getspeed: "<<api.vehicle().getSpeed(mVehicleController->getVehicleId());`
Do you have any explanation regarding this result?

I really appreciate your help,
Kind regards,
Bouchra.

**Reply by riebl** [reply]
I guess that SUMO provides non-zero lateral speed values only if its sublane model is enabled. See https://sumo.dlr.de/docs/Simulation/SublaneModel.html for details.

**Reply by BouchraBenkoussas** [reply]
Hello @riebl,

Thank you for your reply.
Please find attached a picture that describes my problem. I would like to know if the turning angle is the alpha shown in the picture and is it the "Yaw rate value"?
![issue - components](https://user-images.githubusercontent.com/62869736/117574481-8759c380-b0dd-11eb-8a89-fba2e2a91c2d.png)

Kind regards,
Bouchra.

**Reply by riebl** [reply]
The yaw rate is the change of alpha. I assumed that you refer to longitudinal and lateral speed relative to the observed vehicle. However, your picture implies that you are looking for longitudinal and lateral speed relative to the world coordinate system. For the latter, you can simply use the scalar speed, the vehicle's heading and simple trigonometry to derive the vector components.
